packages/flutter/lib/src/painting/border_radius.dart

```diff
@@ -398,12 +400,13 @@ class BorderRadius extends BorderRadiusGeometry {
     // Because the current radii could be negative, we must clamp them before
     // converting them to an RRect to be rendered, since negative radii on
     // RRects don't make sense.
+    final Radius maxRadius = Radius.circular(math.max(rect.width, rect.height));
```

Comment:
Is this right? If the user actually specified an elliptical radius we'll potentially convert it to circular here.

I think this should probably be
```suggestion
    final Radius maxRadius = Radius.elliptical(rect.width, rect.height);
```

Reply:
Also, the comment above should get updated to explain this.

Reply:
I'm going to tell you a secret: 
<img width="577" alt="image" src="https://user-images.githubusercontent.com/351125/226508343-ff15434d-a3dd-498b-9d25-32c22deb8eee.png">

But what you suggested makes more sense than what I did. I'll try that. Thanks!